src/Context/CachedContextFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Roave\BetterReflection\Context;
+
+use Roave\BetterReflection\Context\ContextFactory;
+
+class CachedContextFactory implements ContextFactory
+{
+    private $contexts = [];
+
+    public function __construct(ContextFactory $innerFactory)
+    {
+        $this->innerFactory = $innerFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function createForNamespace($namespace, $fileContents)
+    {
+        $hash = $namespace . md5($fileContents);
```

Comment:
Seems to be the fastest option: https://gist.github.com/dantleech/b442a77dc75c325c83d89b1c2c7db3cb

I assume there will not be any illegal char problems.